### PR TITLE
Breakout sharable sql components.

### PR DESCRIPTION
**Pull request by brandonpollett**
## Description
Describe the changes in this PR.

## Related issues
Addresses [issue #].

## Testing
Describe how this change was tested.

### CODE REVIEW

File: src/Microsoft.Health.SqlServer.Api/Controllers/SchemaController.cs [hunk 1]
```diff
@@ -45,10 +48,10 @@ public ActionResult AvailableVersions()
 
             var availableSchemas = new List<object>();
             var currentVersion = _schemaInformation.Current ?? 0;
-            foreach (var version in Enum.GetValues(typeof(SchemaVersion)).Cast<SchemaVersion>().Where(sv => sv >= currentVersion))
+            foreach (var version in Enum.GetValues(typeof(TSchemaVersionEnum)).Cast<int>().Where(sv => sv >= currentVersion))
```
Comment (brendankowitz) [inline src/Microsoft.Health.SqlServer.Api/Controllers/SchemaController.cs]: A little bit of thinking out loud here, but could this be driven from ISchemaInformation now? i.e. `for(int i = schemaInformation.Current; i <= schemaInformation.Max...)`?

File: src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlConnectionWrapperFactory.cs [hunk 2]
```diff
@@ -4,7 +4,7 @@
 // -------------------------------------------------------------------------------------------------
 
 using EnsureThat;
-using Microsoft.Health.Fhir.SqlServer.Configs;
+using Microsoft.Health.SqlServer.Configs;
```
Comment (jackliums) [inline src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlConnectionWrapperFactory.cs]: Shouldn't these move into the common code too?

File: src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlServerFhirModel.cs [hunk 3]
```diff
@@ -18,9 +18,12 @@
 using Microsoft.Health.Fhir.Core.Exceptions;
 using Microsoft.Health.Fhir.Core.Features.Definition;
 using Microsoft.Health.Fhir.Core.Models;
-using Microsoft.Health.Fhir.SqlServer.Configs;
 using Microsoft.Health.Fhir.SqlServer.Features.Schema;
 using Microsoft.Health.Fhir.SqlServer.Features.Schema.Model;
+using Microsoft.Health.SqlServer.Configs;
```
Comment (jackliums) [inline src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlServerFhirModel.cs]: We need to do similar type of initialization in the DICOM side too (but just initializing the schema), so it would be good to have a common interface which each project could derive and it automatically just works

File: src/Microsoft.Health.Fhir.SqlServer/Registration/FhirServerBuilderSqlServerRegistrationExtensions.cs [hunk 4]
```diff
@@ -6,17 +6,16 @@
...
-using Microsoft.Health.Fhir.SqlServer.Features.Health;
 using Microsoft.Health.Fhir.SqlServer.Features.Schema;
-using Microsoft.Health.Fhir.SqlServer.Features.Schema.Model;
 using Microsoft.Health.Fhir.SqlServer.Features.Search;
 using Microsoft.Health.Fhir.SqlServer.Features.Search.Expressions.Visitors;
 using Microsoft.Health.Fhir.SqlServer.Features.Storage;
+using Microsoft.Health.SqlServer.Api.Registration;
+using Microsoft.Health.SqlServer.Configs;
+using Microsoft.Health.SqlServer.Features.Schema.Model;
+using Microsoft.Health.SqlServer.Registration;
```
Comment (jackliums) [inline src/Microsoft.Health.Fhir.SqlServer/Registration/FhirServerBuilderSqlServerRegistrationExtensions.cs]: nit: Maybe update the name too. it's no longer experimental.

File: src/Microsoft.Health.Fhir.SqlServer/Features/Schema/SchemaInformation.cs [hunk 5]
```diff
@@ -3,20 +3,23 @@
 // Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
 // -------------------------------------------------------------------------------------------------
 
+using System;
+using Microsoft.Health.SqlServer.Features.Schema;
+
 namespace Microsoft.Health.Fhir.SqlServer.Features.Schema
 {
-    public class SchemaInformation
+    public class SchemaInformation : ISchemaInformation
```
Comment (jackliums) [inline src/Microsoft.Health.Fhir.SqlServer/Features/Schema/SchemaInformation.cs]: should this move to common too?

File: src/Microsoft.Health.Fhir.SqlServer/Microsoft.Health.Fhir.SqlServer.csproj [hunk 6]
```diff
@@ -38,21 +31,8 @@
 
   <ItemGroup>
     <ProjectReference Include="..\Microsoft.Health.Fhir.Core\Microsoft.Health.Fhir.Core.csproj" />
-  </ItemGroup>
```
Comment (jackliums) [inline src/Microsoft.Health.Fhir.SqlServer/Microsoft.Health.Fhir.SqlServer.csproj]: is this removal intentional?

File: src/Microsoft.Health.Fhir.SqlServer.UnitTests/IndentedStringBuilderTests.cs [hunk 7]
```diff
@@ -5,6 +5,7 @@
 
 using System;
 using System.Text;
+using Microsoft.Health.SqlServer;
```
Comment (jackliums) [inline src/Microsoft.Health.Fhir.SqlServer.UnitTests/IndentedStringBuilderTests.cs]: Should this test be moved now that IndentedStringBuilder is part of Microsoft.Health.SqlServer

File: src/Microsoft.Health.SqlServer.Api/Controllers/SchemaController.cs [hunk 8]
```diff
@@ -44,11 +49,11 @@ public ActionResult AvailableVersions()
             _logger.LogInformation("Attempting to get available schemas");
 
             var availableSchemas = new List<object>();
-            var currentVersion = _schemaInformation.Current ?? 0;
-            foreach (var version in Enum.GetValues(typeof(SchemaVersion)).Cast<SchemaVersion>().Where(sv => sv >= currentVersion))
+            var currentVersion = _schemaInformation.Current ?? 1;
```
Comment (jackliums) [inline src/Microsoft.Health.SqlServer.Api/Controllers/SchemaController.cs]: is this intentional change?

File: src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlConnectionWrapperFactory.cs (hunk 2 above)
Comment (brandonpollett) [reply src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlConnectionWrapperFactory.cs]: So far this is only a concept that's needed within the fhir-server. We can move it, but that would require a bit more refactoring as the common transaction interfaces live in the fhir-server core.

File: src/Microsoft.Health.Fhir.SqlServer/Features/Schema/SchemaInformation.cs (hunk 5 above)
Comment (brandonpollett) [reply src/Microsoft.Health.Fhir.SqlServer/Features/Schema/SchemaInformation.cs]: Sounds good to me.

File: src/Microsoft.Health.Fhir.SqlServer/Microsoft.Health.Fhir.SqlServer.csproj (hunk 6 above)
Comment (brandonpollett) [reply src/Microsoft.Health.Fhir.SqlServer/Microsoft.Health.Fhir.SqlServer.csproj]: Nope, good catch!

File: src/Microsoft.Health.SqlServer.Api/Controllers/SchemaController.cs (hunk 8 above)
Comment (brandonpollett) [reply src/Microsoft.Health.SqlServer.Api/Controllers/SchemaController.cs]: Yes, because we no longer have access to the enum in this class, we need to initialize with 1 since we're just using the int iterator to build the data.

File: src/Microsoft.Health.Fhir.SqlServer.UnitTests/IndentedStringBuilderTests.cs (hunk 7 above)
Comment (brandonpollett) [reply src/Microsoft.Health.Fhir.SqlServer.UnitTests/IndentedStringBuilderTests.cs]: Yes, along with `EnumerableExtensionsTests.cs`

File: src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlConnectionWrapperFactory.cs (hunk 2 above)
Comment (jackliums) [reply src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlConnectionWrapperFactory.cs]: The DICOM code actually has these registered but that might be because I just copied the code without looking at it closely. Let's not move the code and we can refactor the code in DICOM side when we incorporate the nuget package.

File: src/Microsoft.Health.SqlServer.Api/Controllers/SchemaController.cs [hunk 9]
```diff
@@ -9,30 +9,35 @@
...
-            EnsureArg.IsNotNull(urlResolver, nameof(urlResolver));
+            EnsureArg.IsNotNull(scriptProvider, nameof(scriptProvider));
+            EnsureArg.IsNotNull(urlHelperFactory, nameof(urlHelperFactory));
+            EnsureArg.IsNotNull(actionContextAccessor, nameof(actionContextAccessor));
             EnsureArg.IsNotNull(logger, nameof(logger));
 
             _schemaInformation = schemaInformation;
-            _urlResolver = urlResolver;
+            _scriptProvider = scriptProvider;
+            _urlHelper = urlHelperFactory.GetUrlHelper(actionContextAccessor.ActionContext);
```
Comment (jackliums) [inline src/Microsoft.Health.SqlServer.Api/Controllers/SchemaController.cs]: Looks like my previous comment got lost. Would using the 'ControllerContext' property work here? It derives from `ActionContext`.
